syntax = "proto3";
import "google/protobuf/empty.proto";
import "component_proto/scene_comp.proto";
option cc_generic_services = true;

message ControllerEnterSceneRequest
{
  SceneInfo scene_info = 1;
}

message ControllerLeaveSceneRequest
{
}

message ControllerLeaveSceneAsyncSavePlayerCompleteRequest
{
}

service ControllerSceneServerPlayerService 
{
  rpc EnterScene (ControllerEnterSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveScene (ControllerLeaveSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveSceneAsyncSavePlayerComplete (ControllerLeaveSceneAsyncSavePlayerCompleteRequest) returns (google.protobuf.Empty);
}